### ISSUE TITLE
salt-cloud command uses wrong repo to install minion on RHEL8

### ISSUE THREAD

**Issue by infantvin** [saltstack/salt#56070]
### Description of Issue
Using the command similar to 

salt-cloud m Salt-Testmachines  machine.abc

to spin a new RHEL8 VM in Vsphere by using salt-cloud.

During the installation stage of the salt-minion on the host machine.abc, there is an error saying 
  Error:
        Command 'ssh -t -t -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none  -p 22 root@10.106.17.209 '/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49/deploy.sh -c '"'"'/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49'"'"' stable 2019.3.2'' failed. Exit code: 1

( I have attached the complete trace output of salt-cloud as a text file to this issue).

On checking the VM/minion system that is being created, a quick look at the saltstack.repo file shows this.

[saltstack]
name=SaltStack archive/2019.3.2 Release Channel for RHEL/CentOS $releasever
baseurl=https://repo.saltstack.com/yum/redhat/8/$basearch/archive/2019.3.2/
skip_if_unavailable=True
gpgcheck=1
gpgkey=https://repo.saltstack.com/yum/redhat/8/$basearch/archive/2019.3.2/SALTST
ACK-GPG-KEY.pub
enabled=1
enabled_metadata=1

If you notice the repo URL https://repo.saltstack.com/yum/redhat/ DOES NOT have a subdirectory called 8. This is causing the command to fail.

According to documentation here, http://repo.saltstack.com/#rhel 
the repo for RHEL8 is actually https://repo.saltstack.com/py3/redhat/ 

Hence, the bootstrapping is not pointing the RHEL8 minion to the correct repository URL

### Setup

salt master is RHEL7 with 2019.2.3 of salt and we are creating VMs with the same version of salt-minion.

### Steps to Reproduce Issue

Setup a salt-master with 2019.3.2 and then try to setup a new RHEL8 based VM using the salt-cloud commands as above or any other method.

The command will fail with the error above.

### Versions Report
# salt --versions-report
Salt Version:
           Salt: 2019.2.3

Dependency Versions:
           cffi: 1.13.2
       cherrypy: unknown
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.26.3
        libnacl: Not Installed
       M2Crypto: 0.31.0
           Mako: 0.8.1
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: 3.7.3
         pygit2: 0.26.4
         Python: 2.7.5 (default, Jun 11 2019, 14:33:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: redhat 7.7 Maipo
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-1062.9.1.el7.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 7.7 Maipo

Master and minion are in sync with versions.

**Comment by infantvin** [comment]
Attaching the error message trace of salt-cloud command.
[salt-cloud-rhel8-error.txt](https://github.com/saltstack/salt/files/4159034/salt-cloud-rhel8-error.txt)

**Comment by dhiltonp** [comment]
Thanks for the report!

**Comment by infantvin** [comment]
Hi,

There is something I experimented on and want to share the findings with you.

As I stated previously, the salt-cloud command is failing at this stage:

Error:
Command 'ssh -t -t -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none -p 22 root@10.106.17.209 '/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49/deploy.sh -c '"'"'/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49'"'"' stable 2019.2.3'' failed. Exit code: 1

I decided to retry the failing ssh command after correcting the repo path in the newly created VM.

So, I edited the saltstack.repo file that the bootstrap creates and used the correct paths containing the salt minion required packages.

Sample of corrected /etc/yum.repos.d/saltstack.repo file
[saltstack]
...
baseurl=https://repo.saltstack.com/py3/redhat/8.1/$basearch/archive/2019.2.3/
...
gpgkey=https://repo.saltstack.com/py3/redhat/8.1/$basearch/archive/2019.2.3/SALTSTACK-GPG-KEY.pub


Once this is done, I saved and exited the file and reran the failed ssh command from the master node.
ssh -t -t -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none -p 22 root@10.106.17.209 '/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49/deploy.sh -c '"'"'/tmp/.saltcloud-41bf7c3c-80d8-457a-89c9-5b92b30c5e49'"'"' stable 2019.2.3

This worked and the minion install on RHEL8 completed successfully.

Snippet of the command execution final summary
---------------------------------------------------
.....
Installed:
  salt-minion-2019.2.3-1.el8.noarch                                             
  libsodium-1.0.18-2.el8.x86_64                                                 
  libunwind-1.3.1-3.el8.x86_64                                                  
  openpgm-5.2.122-21.el8.x86_64                                                 
  python3-m2crypto-0.35.2-5.el8.x86_64                                          
  python3-msgpack-0.6.2-1.el8.x86_64                                            
  python3-psutil-5.6.3-5.el8.x86_64                                             
  zeromq-4.3.2-1.el8.x86_64                                                     
  python3-markupsafe-0.23-19.el8.x86_64                                         
  python3-jinja2-2.10.1-2.el8_0.noarch                                          
  python3-distro-1.4.0-2.module+el8.1.0+3334+5cb623d7.noarch                    
  python3-babel-2.5.1-5.el8.noarch                                              
  python3-tornado4-4.5.2-3.el8.x86_64                                           
  python3-zmq-17.0.0-5.el8.x86_64                                               
  salt-2019.2.3-1.el8.noarch                                                    

Complete!
*  INFO: Running install_red_hat_enterprise_linux_stable_post()
*  INFO: Running install_red_hat_enterprise_linux_restart_daemons()
*  INFO: Running daemons_running()
*  INFO: Salt installed!
----------------------------------------

So, it looks like whatever salt-cloud calls to stage the repos and install the minion will need to be changed to point to the correct URL for the saltstack.repo used in RHEL8.

Hope that this information will be useful to you.

We are awaiting a fix so that we can start standardizing RHEL8 for our product build environment using saltstack rules we have developed over the years.

Thanks.

**Comment by sagetherage** [comment]
@infantvin have you updated to a different version of RHEL and still seeing this issue?

**Comment by sagetherage** [comment]
@infantvin have you updated to a different version of RHEL and still seeing this issue?

Reviewed in Grooming 2021-JUL-19 considering closing

**Comment by Ch3LL** [comment]
Closing due to inactivity. If this is still occurring on the latest version of salt please open a new issue with the requested information and details.
